Sprint 9 solution http api

shepilovaElena opened the pull request.

shepilovaElena committed:
тз-8.
check.

shepilovaElena committed:
тз-9.
check.

shepilovaElena committed:
тз-9.
классы HttpTaskServer, LocalDateTimeAdapter, хэндлеры.

shepilovaElena committed:
тз-9.
Правки хендлеров.
Тесты.

shepilovaElena committed:
тз-9.
класс Main.

shepilovaElena committed:
тз-9.
Правки InMemoryHistoryTaskManagerTest.

shepilovaElena committed:
тз-9.
класс DurationAdapter.

shepilovaElena committed:
тз-9.
code style.

shepilovaElena committed:
тз-9.
правки в работе хендлеров.

irek-valiev commented on src/ru/yandex/practicum/javakanban/handlers/EpicsHandler.java:
```diff
@@ -0,0 +1,126 @@
+                    sendNotFound(exchange, "ошибка");
+                    break;
+                }
+
```
в задании для Эпиков на метод POST есть баг. Ну нужно реализовывать по аналогии с задачами/подзадачами, что если указан id, то нужно обновлять, а если нет id, то создавать

irek-valiev commented on src/ru/yandex/practicum/javakanban/handlers/SubtasksHandler.java:
```diff
@@ -0,0 +1,131 @@
+                            sendHasInteractions(exchange, "Задача пересекаются по времени с уже существующими.");
+                            break;
+                        } else {
+                            sendText201(exchange, "Действие выполнено корректно.");
```
после создания новой сущности, в ответе нужно сообщить присвоенное ей id, по-другому "на том конце провода" его никак не узнают

irek-valiev commented on src/ru/yandex/practicum/javakanban/handlers/TasksHandler.java:
```diff
@@ -0,0 +1,127 @@
+                            sendHasInteractions(exchange, "Задача пересекаются по времени с уже существующими.");
+                            break;
+                        } else {
+                            sendText201(exchange, "Действие выполнено корректно.");
```
и тут тоже

irek-valiev commented on src/Main.java:
```diff
@@ -1,9 +1,49 @@
+
+    private static final int PORT = 8080;
+    private static InMemoryTaskManager taskManager;
+    private static HistoryManager historyManager;
```
Ты эту переменную нигде не используешь, ее стоит удалить

irek-valiev commented on src/Main.java:
```diff
@@ -1,9 +1,49 @@
+
+    public static void main(String[] args) throws IOException {
+
+        HttpServer httpServer = HttpServer.create(new InetSocketAddress(8080), 0);
```
создание сервера и поднятие контекста не реализуется на стороне клиента (а main в нашем случае играет роль клиента). Нужно добавить класс, типа TaskManagerServer где будут как раз контекст и методы старта и стопа сервера. А в маине только использовать этот класс и методы

irek-valiev commented on src/Main.java:
```diff
@@ -1,9 +1,49 @@
+        httpServer.createContext("/prioritized", new PrioritizedHandler(taskManager));
+        startServer(httpServer);
+
+        System.out.println("HTTP-сервер запущен на " + PORT + " порту!");
```
это скорее нужно в методе startServer()

shepilovaElena committed:
тз-9.
создание класса HttpTaskServer.
правки метода POST в хендлерах.
правки в EpicsHandler.

shepilovaElena committed:
тз-9.
удаление ненужных импортов.

shepilovaElena committed:
тз-9.
удаление ненужных импортов.
